src/common/src/array/arrow/arrow_impl.rs

```diff
@@ -524,9 +524,12 @@ pub trait FromArrow {
             Float64 => self.from_float64_array(array.as_any().downcast_ref().unwrap()),
             Date32 => self.from_date32_array(array.as_any().downcast_ref().unwrap()),
             Time64(Microsecond) => self.from_time64us_array(array.as_any().downcast_ref().unwrap()),
-            Timestamp(Microsecond, _) => {
+            Timestamp(Microsecond, None) => {
                 self.from_timestampus_array(array.as_any().downcast_ref().unwrap())
             }
+            Timestamp(Microsecond, Some(_)) => {
+                self.from_timestampus_some_array(array.as_any().downcast_ref().unwrap())
+            }
```

Comment:
Why do we need this change?

Reply:
This may be a bug that we implement `from_field(&self, field: &arrow_schema::Field) -> Result<DataType, ArrayError>`, where we treat  `Timestamp(Microsecond, None)` and `Timestamp(Microsecond, Some(_))` differently into Timestamp and Timestamptz(refer to https://github.com/risingwavelabs/risingwave/blob/d5fc35c61a7384a62499c309a114e2efa63662b5/src/common/src/array/arrow/arrow_impl.rs#L460-L461), while we only implement `from Timestamp(Microsecond, _)`(refer to https://github.com/risingwavelabs/risingwave/blob/d5fc35c61a7384a62499c309a114e2efa63662b5/src/common/src/array/arrow/arrow_impl.rs#L527-L528). 

This means if the type we define is `Timestamptz`, it will not be parsed correctly, so I fixed it. cc @wangrunji0408